src/nisarqa/utils/metadata_checks.py

```diff
@@ -168,102 +166,204 @@ def verify_metadata_cubes(
+        the z-dimension height layers in a 3D dataset has all non-finite values.
     """
 
-    # Flag indicating if metadata cubes pass all verification checks; used
+    # Flag indicating if metadata datasets pass all verification checks; used
     # for Summary CSV reporting
     passes = True
     has_finite = True
 
-    # helper function to check if the cube is GDAL-friendly.
-    # no-op if the cube is not geocoded and/or not an h5py.Dataset
-    def _check_gdal(c: nisarqa.MetadataCube2D | nisarqa.MetadataCube3D) -> bool:
-        if product.is_geocoded and isinstance(c.data, h5py.Dataset):
-            return is_gdal_friendly(
-                input_filepath=product.filepath, ds_path=c.data.name
-            )
-        else:
-            return True
-
-    # Check metadata cubes in metadata Group
+    # Check metadata datasets in metadata Group
     try:
-        # Note: During the __post_init__ of constructing each MetadataCube,
+        # Note: During the __post_init__ of constructing each metadata dataset,
         # several validation checks are performed, including ensuring that
         # there are corresponding datasets with x coordinates and y coordinates
         # of the correct length. If these elements are missing, exceptions
         # will get thrown.
         for cube in product.coordinate_grid_metadata_cubes():
-            has_finite &= _metadata_cube_has_finite_pixels(cube)
+            has_finite &= _dataset_has_finite_pixels(cube)
             passes &= has_finite
-            passes &= _metadata_cube_is_not_all_zeros(cube)
-            passes &= _check_gdal(c=cube)
+            passes &= _dataset_is_not_all_zeros(cube)
+            passes &= _check_gdal(product=product, ds=cube)
+
+    except (nisarqa.DatasetNotFoundError, ValueError) as e:
+        nisarqa.get_logger().error(traceback.format_exc())
+        passes = False
+
+    # SUMMARY LOG
+    summary = nisarqa.get_summary()
+    summary.check_metadata_cubes(result="PASS" if passes else "FAIL")
+
+    if fail_if_all_nan and (not has_finite):
+        raise nisarqa.InvalidRasterError(
+            "One or more metadata cubes contains all non-finite"
+            " (e.g. NaN, +/- Inf) values or one of the z-dimension height"
+            " layers in a 3D cube has all non-finite values. See log file"
+            " for details and names of the failing dataset(s)."
+        )
+
+
+def verify_calibration_metadata(
+    product: nisarqa.NonInsarProduct, fail_if_all_nan: bool = True
+) -> None:
+    """
+    Verify if the input product's calibration metadata datasets are valid.
+
+    Parameters
+    ----------
+    product : nisarqa.NisarNonInsarProductProduct
+        Instance of the input product.
+    fail_if_all_nan : bool, optional
+        True to raise an exception if one of the metadata datasets contains
+        all non-finite (e.g. Nan, +/- Inf) values, or if one of the
+        z-dimension height layers in a 3D dataset has all non-finite values.
```

Comment:
To my knowledge, the calibration metadata LUTs are all either 1-D or 2-D.

```suggestion
        all non-finite (e.g. Nan, +/- Inf) values.
```